List/LocationList.py

```diff
@@ -84,43 +84,43 @@ def getTreasureList():
-            KH2Treasure(113, "Spooky Cave Guard Recipe", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE]),
-            KH2Treasure(115, "Spooky Cave Mythril Crystal", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE]),
-            KH2Treasure(116, "Spooky Cave AP Boost", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE]),
-            KH2Treasure(284, "Sweet Memories", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
-            KH2Treasure(485, "Spooky Cave Map", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
-            KH2Treasure(312, "Starry Hill Cosmic Ring", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE]),
-            KH2Treasure(94, "Starry Hill Style Recipe", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE]),
-            KH2Treasure(285, "Starry Hill Cure Element", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
-            KH2Treasure(539, "Starry Hill Orichalcum+", LocationTypes=[locationType.HUNDREDAW], InvalidChecks=[itemType.TORN_PAGE, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
-
-            KH2Treasure(7,"Passage Mythril Shard", LocationTypes=[locationType.OC]),
-            KH2Treasure(8,"Passage Mythril Stone", LocationTypes=[locationType.OC]),
-            KH2Treasure(144, "Passage Ether", LocationTypes=[locationType.OC]),
-            KH2Treasure(145, "Passage AP Boost", LocationTypes=[locationType.OC]),
-            KH2Treasure(146, "Passage Hi-Potion", LocationTypes=[locationType.OC]),
-            KH2Treasure(2,"Inner Chamber Underworld Map", LocationTypes=[locationType.OC]),
-            KH2Treasure(243, "Inner Chamber Mythril Shard", LocationTypes=[locationType.OC]),
+            KH2Treasure(548, "Marluxia (AS) Eternal Blossom", LocationTypes=[locationType.DC, locationType.AS], InvalidChecks=[itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(553, "Marluxia (Data) Lost Illusion", LocationTypes=[locationType.DC, locationType.AS], InvalidChecks=[itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(587, "Lingering Will Proof of Connection", LocationTypes=[locationType.DC, locationType.LW], InvalidChecks=([] if trueRandom else [itemType.FORM, itemType.THUNDER, itemType.MAGNET, itemType.TORN_PAGE])+[itemType.PROOF_OF_CONNECTION, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
```

Comment:
I'm confused by your first comment. Are you suggesting checking what type of item placement logic is in use when trying to assign an item to a location? I believe that would just turn into having a different dictionary of location ids and replacement `InvalidChecks` lists to overwrite with. That would make the item placement portion do a check every time to see if the location is in the override list, and if so, use the modified list, and otherwise, use the original. Doing it this way makes the check done once, and if we need to reassign items due to a conflict, we don't redo the checks. 

I agree about the popup thing, but the always invalid things are not just related to being a popup, such as Lingering will not allowing to hold the proof of connection. That should always be included and would still require the conditional statement to handle even if the popup distinction was made. It would clean things up a bit though, so maybe in a follow-up, you or I could do that.